lib/db/mem.js

```diff
@@ -1034,20 +1047,34 @@ module.exports = function (log, error) {
     // Also include email in accounts table
     return getAccountByUid(uid)
       .then(function (account) {
-        userEmails.push({
-          email: account.email,
-          isPrimary: true,
-          isVerified: account.emailVerified
-        })
-
-        // Return all emails in emails table
+        var containsPrimaryEmail = false
         Object.keys(emails).forEach(function (key) {
           var emailRecord = emails[key]
           if (emailRecord.uid.toString('hex') === uid.toString('hex')) {
+            if (emailRecord.isPrimary) {
+              containsPrimaryEmail = true
+            }
             userEmails.push(emailRecord)
           }
         })
-        return userEmails
+
+        if (! containsPrimaryEmail || userEmails.length === 0) {
```

Comment:
Yea, that is correct. Since an email is always added to the emails table, it is safe to remove.